modules/claims_api/app/controllers/concerns/claims_api/disability_compensation_validations.rb

```diff
@@ -411,30 +411,44 @@ def validate_form_526_disability_secondary_disabilities!
     end
 
     def validate_form_526_disability_classification_code!
-      return if (form_attributes['disabilities'].pluck('classificationCode') - [nil]).blank?
+      form_attributes['disabilities'].each_with_index do |disability, index|
+        classification_code = disability['classificationCode']
+        next if classification_code.nil? || classification_code.blank?
+
+        if bgs_classification_ids.include?(classification_code)
+          validate_form_526_disability_classification_code_end_date!(classification_code, index)
+        else
+          raise ::Common::Exceptions::InvalidFieldValue.new("disabilities.#{index}.classificationCode",
+                                                            classification_code)
+        end
+      end
+    end
 
-      form_attributes['disabilities'].each do |disability|
-        next if disability['classificationCode'].blank?
-        next if bgs_classification_ids.include?(disability['classificationCode'])
+    def validate_form_526_disability_classification_code_end_date!(classification_code, index)
+      bgs_disability = contention_classification_type_code_list.find { |d| d[:clsfcn_id] == classification_code }
+      end_date = bgs_disability[:end_dt] if bgs_disability
 
-        raise ::Common::Exceptions::InvalidFieldValue.new('disabilities.classificationCode',
-                                                          disability['classificationCode'])
-      end
+      return if end_date.nil?
+
+      return if Date.parse(end_date) >= Time.zone.today
+
+      raise ::Common::Exceptions::InvalidFieldValue.new("disabilities.#{index}.classificationCode", classification_code)
```

Comment:
I think we'll want to clarify that it's due to an expiration of the Classification Code? or are we good with just saying "nope". @klwatanabe - thoughts? 
```
"\"10\" is not a valid value for \"disabilities.0.classificationCode\""
```

Calling this out - I don't think the common::exceptions::invalid has an easy way to ingest a more meaningful description though so maybe this is an additional ticket / ask if so ?

Reply:
I agree that it would be a better user experience to return the added detail that it's not valid because it is expired. As you noted, we could track that in a separate ticket since our current error handling does not support that.

I will also note (for priority context) that based on the new logging we added, it looks like our v1 consumers are not submitting classification codes today.

Reply:
Agreed - separate ticket would be best, so we can address the error handling more broadly.